cmd/nerdctl/builder_build_test.go

```diff
@@ -592,3 +592,58 @@ func TestBuildAttestation(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestBuildWithPull(t *testing.T) {
+	testutil.RequiresBuild(t)
+
+	oldImage := "ubuntu:18.04"
```

Comment:
Any reason to use this very old version of Ubuntu?

Reply:
Just for testing purposes. The idea is to pull an old version of ubuntu, tag it as latest, and see if `--pull=false` uses local image while `--pull=true` uses the remote image (latest vesion).

Reply:
Can we just use the mirror registry, with updating the tag with another image before checking --pull behaviror?

Reply:
Makes sense, I switched to using much slimmer images as well so this test will now run a little bit faster